src/app/modules/checkout/components/line-items/line-items.component.ts

```diff
@@ -12,11 +15,64 @@ import { Cart } from 'src/app/models/cart.model';
 export class LineItemsComponent {
   @Input() cart?: Cart;
 
+  constructor(
+    private readonly store: Store
+  ) { }
+
+  private readonly typesToCheck: string[] = [
+    'Integer',
+    'ShortText',
+    'DateTime',
+  ];
+
+  getDynamicPropertiesToDisplay(
+    props: ReadonlyArray<(cart_cart_items_dynamicProperties | null)> | null
+  ): DisplayItem[] {
+    if (props == null) {
+      return [];
+    }
+
+    return props.filter(p => this.typesToCheck.includes(p?.valueType ?? '') && p?.value || p?.valueType === 'Boolean')
+      .map(p => {
+        let displayValue = '';
+
+        switch (p?.valueType) {
+        case 'Boolean':
+          displayValue = p?.value && p.value.toLowerCase() === 'true' ? 'yes' : 'no';
+          break;
+
+        case 'DateTime':
+          if (p?.value) {
+            displayValue = new Date(p.value).toLocaleDateString();
+          }
+          break;
+
+        default:
+          displayValue = p?.value ?? '';
+          break;
+        }
+
+        return {
+          name: p?.name,
+          displayValue,
+        };
+      });
+  }
+
```

Comment:
Shouldn't we introduce something like service for dynamic properties conversion?
Also, may be it possible to return typed value for dynamic property? Or introduce special type for them. Something like
```typescript
type DP = { valueType: 'Integer', value: number } | { valueType: 'DateTime', value: Date }
```

Reply:
TODO: add dynamic property component containing all the logic. But only when the code would be reused in yet another place (avoid duplication).